pytorch_lightning/accelerators/ddp_backend.py

```diff
@@ -123,14 +139,116 @@ def __ddp_script_mode_setup(self):
+            return self.ddp_train(process_idx=self.task_idx, mp_queue=None, model=model)
+
+    def ddp_train(self, process_idx, mp_queue, model, is_master=False, proc_offset=0):
+        """
+        Entry point for ddp
+        Args:
+            process_idx:
+            mp_queue: multiprocessing queue
+            model:
+            is_master:
+            proc_offset:
+        Returns:
+        """
+        # offset the process id if requested
+        process_idx = process_idx + proc_offset
+
+        # show progressbar only on progress_rank 0
+        if (self.trainer.node_rank != 0 or process_idx != 0) and self.trainer.progress_bar_callback is not None:
+            self.trainer.progress_bar_callback.disable()
+
+        # determine which process we are and world size
+        self.set_world_ranks(process_idx)
+
+        # set warning rank
+        rank_zero_only.rank = self.trainer.global_rank
+
+        # set up server using proc 0's ip address
+        # try to init for 20 times at max in case ports are taken
+        # where to store ip_table
+        model.trainer = self.trainer
+        model.init_ddp_connection(
+            self.trainer.global_rank,
+            self.trainer.world_size,
+            self.trainer.is_slurm_managing_tasks
+        )
+
+        # call setup after the ddp process has connected
+        self.trainer.call_setup_hook(model)
+
+        # on world_size=0 let everyone know training is starting
+        if self.trainer.is_global_zero and not torch.distributed.is_initialized():
+            log.info('-' * 100)
+            log.info(f'distributed_backend={self.trainer.distributed_backend}')
+            log.info(f'All DDP processes registered. Starting ddp with {self.trainer.world_size} processes')
+            log.info('-' * 100)
+
+            # call sync_bn before .cuda(), configure_apex and configure_ddp
+            if self.trainer.sync_batchnorm:
+                model = model.configure_sync_batchnorm(model)
+
+        # MODEL
+        # copy model to each gpu
+        self.model_to_device(model, process_idx, is_master)
+
+        # CHOOSE OPTIMIZER
+        # allow for lr schedulers as well
+        self.setup_optimizers(model)
+
+        # set model properties before going into wrapper
+        self.trainer.model_connector.copy_trainer_model_properties(model)
+
+        # AMP - run through amp wrapper before going to distributed DP
+        # DDP uses all GPUs on the machine
+        device_ids = self.get_device_ids()
+
+        # allow user to configure ddp
+        model = model.configure_ddp(model, device_ids)
+
+        # set up training routine
+        self.barrier('ddp_setup')
+        self.trainer.train_loop.setup_training(model)
+
+        # train or test
+        results = self.train_or_test()
+
+        # clean up memory
+        torch.cuda.empty_cache()
+
+        return results
+
+    def training_step(self, args):
```

Comment:
```suggestion
    def training_step(self, *args):
```

changing this to positional args allows for inspection. Since this is also passed as positional args, this should be okay. Just have to check the according calls as well